core/drivers/fabric-driver/server/listener.ts

```diff
@@ -0,0 +1,198 @@
+            const payloadData = Object.values(payload)[Object.keys(payload).indexOf('data')];
+            const transactions = payloadData.actions;
+            // Iterate through the transaction list
+            transactions.forEach(async (transaction: any) => {
+                // Get transaction chaincode ID
+                const chaincodeId = transaction.payload.chaincode_proposal_payload.input.chaincode_spec.chaincode_id.name;
+                if (transaction.payload.chaincode_proposal_payload.input.chaincode_spec.input.args.length > 0) {
+                    // Get transaction function name: first argument according to convention
+                    const chaincodeFunc = transaction.payload.chaincode_proposal_payload.input.chaincode_spec.input.args[0].toString();
+                    console.log('Trying to find match for channel', channelId, 'chaincode', chaincodeId, 'function', chaincodeFunc);
+                    // Find all matching event subscriptions stored in the database
+                    let eventMatcher = new events_pb.EventMatcher();
+                    eventMatcher.setEventType(events_pb.EventType.LEDGER_STATE);
+                    eventMatcher.setEventClassId('');       // Only match subscriptions where class ID is not specified
+                    eventMatcher.setTransactionLedgerId(channelId);
+                    eventMatcher.setTransactionContractId(chaincodeId);
+                    eventMatcher.setTransactionFunc(chaincodeFunc);
+                    let eventSubscriptionQueries;
+                    for (let i = 0 ; i < 10 ; i++) {
+                        try{
+                            eventSubscriptionQueries = await lookupEventSubscriptions(eventMatcher);
+                            break;
+                        } catch(error) {
+                            let errorString: string = error.toString();
+                            if (!errorString.includes('LEVEL_LOCKED')) {    // Check if contention was preventing DB access
+                                throw(error);
+                            }
+                            await new Promise(f => setTimeout(f, 2000));    // Sleep 2 seconds
+                        }
+                    }
+                    // Iterate through the view requests in the matching event subscriptions
+                    eventSubscriptionQueries.forEach(async (eventSubscriptionQuery: query_pb.Query) => {
+                        console.log('Generating view and collecting proof for channel', channelId, 'chaincode', chaincodeId);
+                        // Trigger proof collection
+                        const [result, invokeError] = await handlePromise(
+                            invoke(
+                                eventSubscriptionQuery,
+                                networkName,
+                            ),
+                        );
+                        if (!invokeError) {
```

Comment:
I'm just wondering if there is an error should be sending that error message also to the subscriber?
Because Event occurred but due to error in query, we are not sending anything, what is something is wrong in query view address itself, and that needs to be rectified?

Reply:
In the case you mention, yes, it'd be useful to send some feedback to the subscriber. But chances are most of these errors are transient ones, local to the source network, and we may end up spamming the subscriber. And also, if there is something wrong with the view address, the matching logic in the event listener is not going to catch it, and will instead silently ignore all block/contract events.

Error management is something we can design for later? For now, let event publish be best-effort, and let us assume that it is the responsibility of the subscriber to do its due diligence before sending a subscription request.